src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/AuthorizationRule/SetAzureEventhubAuthorizationRules.cs

```diff
@@ -21,86 +21,48 @@
         [ValidateNotNullOrEmpty]
         [Alias(AliasEventHubName)]
         public string EventHub { get; set; }
 
-        [Parameter(Mandatory = true,
-            ValueFromPipelineByPropertyName = true,
-            Position = 3,
-            HelpMessage = "AuthorizationRule Name.")]
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 3, HelpMessage = "AuthorizationRule Name")]
         [ValidateNotNullOrEmpty]
         [Alias(AliasAuthorizationRuleName)]
         public string Name { get; set; }
 
-        [Parameter(Mandatory = true,
-            ValueFromPipeline = true,
-            Position = 4,
-            ParameterSetName = AuthoRuleInputObjectParameterSet,
-            HelpMessage = "Relay AuthorizationRule Object.")]
-        [Parameter(Mandatory = false,
-            ValueFromPipeline = true,
-            Position = 4,
-            ParameterSetName = NamespaceAuthoRuleParameterSet,
-            HelpMessage = "Relay AuthorizationRule Object.")]
-        [Parameter(Mandatory = false,
-            ValueFromPipeline = true,
-            Position = 4,
-            ParameterSetName = EventhubAuthoRuleParameterSet,
-            HelpMessage = "Relay AuthorizationRule Object.")]
+        [Parameter(Mandatory = true, ParameterSetName = AuthoRuleInputObjectParameterSet, ValueFromPipeline = true, Position = 4, HelpMessage = "AuthorizationRule Object")]
+        [Parameter(Mandatory = false, ParameterSetName = NamespaceAuthoRuleParameterSet, ValueFromPipeline = true, Position = 4, HelpMessage = "AuthorizationRule Object")]
+        [Parameter(Mandatory = false, ParameterSetName = EventhubAuthoRuleParameterSet, ValueFromPipeline = true, Position = 4, HelpMessage = "AuthorizationRule Object")]
         [ValidateNotNullOrEmpty]
         [Alias(AliasAuthRuleObj)]
-        public SharedAccessAuthorizationRuleAttributes InputObject { get; set; }
+        public PSSharedAccessAuthorizationRuleAttributes InputObject { get; set; }
 
-        [Parameter(Mandatory = true,
-            Position = 4,
-            ParameterSetName = AuthoRulePropertiesParameterSet,
-            ValueFromPipelineByPropertyName = true,
-            HelpMessage = "Rights, e.g.  @(\"Listen\",\"Send\",\"Manage\")")]
-        [Parameter(Mandatory = false,
-            Position = 4,
-            ParameterSetName = NamespaceAuthoRuleParameterSet,
-            ValueFromPipelineByPropertyName = true,
-            HelpMessage = "Rights, e.g.  @(\"Listen\",\"Send\",\"Manage\")")]
-        [Parameter(Mandatory = false,
-            Position = 4,
-            ParameterSetName = EventhubAuthoRuleParameterSet,
-            ValueFromPipelineByPropertyName = true,
-            HelpMessage = "Rights, e.g.  @(\"Listen\",\"Send\",\"Manage\")")]
+        [Parameter(Mandatory = true, ParameterSetName = AuthoRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 4, HelpMessage = "Rights, e.g.  @(\"Listen\",\"Send\",\"Manage\")")]
```

Comment:
implemented if block for 'AuthoRuleInputObjectParameterSet' and seems like 'AuthoRulePropertiesParameterSet' is not required. will need to remove this in breaking change

Reply:
If the AuthoRulePropertiesParameterSet never worked, it would not be a breaking change to remove it.  If you believe it should be removed, remove it here.

Reply:
If the AuthoRulePropertiesParameterSet never worked, it would not be a breaking change to remove it. If you believe it should be removed, remove it here.